securedrop_client/gui/widgets.py

```diff
@@ -1862,6 +1888,47 @@ def set_failed_to_decrypt_styles(self) -> None:
         self.color_bar.setStyleSheet(self.STATUS_BAR_CSS)
         self.sender_icon.set_failed_to_decrypt_styles()
 
+    def update_seen_by_list(self, usernames: Dict[str, User]) -> None:
+        # Update the dictionary for the new usernames to be shown in the tooltip.
+        self.seen_by.update(usernames)
```

Comment:
The last remaining work is to handled deleted users. Currently, this will only add new usernames to the seen_by dictionary. `SpeechBubble.seen_by` will need to be also remove usernames that are not supplied via the `usernames` dictionary. (Just as a reminder, the "deleted" user will be provided in the supplied `usernames` dictionary if one or more users have been deleted, so you don't have to worry about representing the deleted user.)  Either as part of this PR or as followup, I would recommend changing the dictionary from tracking usernames to tracking UUIDs. For now, it's fine to do a username check since we know that the usernames are required to be unique by the server.